src/Objects/MonitorManager.vala

```diff
@@ -234,6 +234,24 @@ public class Display.MonitorManager : GLib.Object {
             }
         }
 
+        int min_x = int.MAX;
+        int min_y = int.MAX;
+
+        // Make sure the remaining enabled monitors start at 0,0 as required by mutter.
+        // Calculate their offset, and then offset them if necessary
+        foreach (unowned var logical_monitor in logical_monitors) {
+            min_x = int.min (min_x, logical_monitor.x);
+            min_y = int.min (min_y, logical_monitor.y);
+        }
+
+        if (min_x != 0 || min_y != 0) {
+            // Iterate over the items in the array directly here, a `foreach` would return a copy
+            for (int i = 0; i < logical_monitors.length; i++) {
+                logical_monitors[i].x -= min_x;
+                logical_monitors[i].y -= min_y;
```

Comment:
I believe this would work
```suggestion
            foreach (unowned var logical_monitor in logical_monitors) {
                logical_monitor.x -= min_x;
                logical_monitor.y -= min_y;
```

Reply:
Just tested this, the values in the array passed to the mutter DBus are unmodified after doing this. So by doing this, you're not modifying the instance of the struct in the array. So, must still be a copy happening somewhere.